### PR TITLE
[TTAHUB-1827] "View" training reports

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
- Add a way to view training and session reports in a read only mode at ```/training-report/view/${id}```
- Added the "View event" button to the training reports table
- Also added two hooks to the backend for sessionReports. One that automatically moves a "not started" training report to "in progress" a session is created for it. The other prevents changing or deleting a session if the training report is complete.

## How to test
- Fill out a training report and some session reports
- Visit the training report view link and make sure it looks correct and matches the mockup
- Tests out the hooks - add a session to a training report and confirm the status is updated. Destroy or update a session on a completed training report.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1827

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/VanillaModal.js [hunk 1]
```diff
@@ -8,7 +8,10 @@ export default function VanillaModal({
   modalRef,
   className,
 }) {
-  const modalId = `modal-${heading.toLowerCase().replace(/\s/g, '-')}`;
+  // strip everything but letters
+  const regex = /[^a-zA-Z]/g;
```
Comment (AdamAdHocTeam) [inline frontend/src/components/VanillaModal.js:12]: All I see are vulnerabilities 🤣

File: frontend/src/components/renderData.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
```
Comment (AdamAdHocTeam) [inline frontend/src/components/renderData.js:1]: Can you give this a more meaningful name given it lives in components (more context),

File: frontend/src/pages/ApprovedActivityReport/components/ApprovedReportV2.js [hunk 3]
```diff
@@ -254,7 +254,7 @@ export default function ApprovedReportV2({ data }) {
           : null }
       </div>
 
-      <ApprovedReportSection
+      <ReadOnlyContent
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/ApprovedActivityReport/components/ApprovedReportV2.js:257]: Nice generic conversion.

File: frontend/src/pages/TrainingReports/components/EventCard.js [hunk 4]
```diff
@@ -24,7 +26,7 @@ function EventCard({
   const contextMenuLabel = `Actions for event ${event.id}`;
   const menuItems = [];
 
-  if (event.status !== 'Complete') {
+  if (data.status !== TRAINING_REPORT_STATUSES.COMPLETE) {
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/TrainingReports/components/EventCard.js:29]: Good call.

File: src/models/hooks/sessionReportPilot.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
+const { Op } = require('sequelize');
+const { TRAINING_REPORT_STATUSES } = require('@ttahub/common');
+const { auditLogger } = require('../../logger');
+
+const preventChangesIfEventComplete = async (sequelize, instance, options) => {
```
Comment (AdamAdHocTeam) [inline src/models/hooks/sessionReportPilot.js:5]: Are these like BE fail safes?

**Review by AdamAdHocTeam — COMMENTED** [review]
As we discussed lets fix the language to say the drop down must say 'complete'.

image

File: src/models/hooks/sessionReportPilot.js (hunk 5 above)
Comment (thewatermethod) [reply src/models/hooks/sessionReportPilot.js:5]: Yeah

File: frontend/src/components/renderData.js (hunk 2 above)
Comment (thewatermethod) [reply frontend/src/components/renderData.js]: https://github.com/HHS/Head-Start-TTADP/pull/1598/commits/679d64a2933b598f8cd7837bb4e7aaa55ae6cc94

**Comment by thewatermethod** [comment]
> As we discussed lets fix the language to say the drop down must say 'complete'.

I have updated this 
https://github.com/HHS/Head-Start-TTADP/pull/1598/commits/679d64a2933b598f8cd7837bb4e7aaa55ae6cc94

**Review by AdamAdHocTeam — APPROVED** [review]
Overall LGTM!

Left some minor comments / suggestions.

**Pull request merged**